static/js/components/subject_page/main_pane.tsx

```diff
@@ -112,30 +113,38 @@ export const SubjectPageMainPane = memo(function SubjectPageMainPane(
 
   return (
     <div id="subject-page-main-pane">
-      {!_.isEmpty(props.pageConfig) &&
-        !_.isEmpty(props.pageConfig.categories) &&
-        props.pageConfig.categories.map((category, idx) => {
-          const id = getId(props.id, CATEGORY_ID_PREFIX, idx);
-          return (
-            <DataContext.Provider key={id} value={contextData}>
-              <ErrorBoundary>
-                <Category
-                  id={id}
-                  place={props.place}
-                  enclosedPlaceType={enclosedPlaceType}
-                  config={category}
-                  eventTypeSpec={props.pageConfig.metadata.eventTypeSpec}
-                  svgChartHeight={
-                    // TODO: Unroll this for NL and use container offset from CSS instead.
-                    props.svgChartHeight
-                      ? props.svgChartHeight
-                      : SVG_CHART_HEIGHT
-                  }
-                />
-              </ErrorBoundary>
-            </DataContext.Provider>
-          );
-        })}
+      <DataFetchContextProvider id={props.id}>
+        {!_.isEmpty(props.pageConfig) &&
+          !_.isEmpty(props.pageConfig.categories) &&
+          props.pageConfig.categories.map((category, idx) => {
+            const id = getId(props.id, CATEGORY_ID_PREFIX, idx);
+            return (
+              <DataContext.Provider
```

Comment:
could you add a todo here to combine this with DataFetchContextProvider? it could be used with some constant keys.. at the moment, it's hard to know from naming how to differentiate between the two

Reply:
done